ouroboros-network-framework/src/Simulation/Network/Snocket.hs

```diff
@@ -204,6 +206,15 @@ data NetworkState m addr = NetworkState {
 
     }
 
+-- | Simulation accessible network environment consumed by 'simSnocket'.
+--
+data UniverseState addr = UniverseState {
+      -- | Registry of active connections and respective provider
+      --
+      usConnections       :: Map (NormalisedId addr) addr
```

Comment:
Can we use a newtype for the exposed connection state?  It will be easier to extend in the future.